lib/pbench/test/functional/agent/cli/commands/test_config.py

```diff
@@ -2,15 +2,17 @@
 import pathlib
 import pytest
 
+from pbench.agent.base import BaseCommand
+
 
 def test_pbench_run(monkeypatch, tmpdir, agent_config, pbench_run):  # noqa F811
     """Use pbench_run value from pbench-agent.cfg file"""
     monkeypatch.delenv("pbench_run")
     env_pbench_run = os.environ.get("pbench_run")
+    assert env_pbench_run is None
     command = ["pbench-list-tools"]
     out, err, exitcode = pytest.helpers.capture(command)
     assert exitcode == 0
-    assert env_pbench_run is None
     assert pathlib.Path(f"{tmpdir}/var/lib/pbench-agent") == pbench_run
```

Comment:
Does this assertion (at line 16) actually test anything relevant to this test?  It appears to be checking that the value the `pbench_run` _fixture_ is what we expect it to be...which has nothing to do with the execution of `pbench-list-tools` nor with the changes made in this PR.  (This is the problem of _functional_ testing.)

If you're able to set this up as a unit test (see earlier [comment](https://github.com/distributed-system-analysis/pbench/pull/2606#discussion_r803063752)), then I think you can make this assertion test something meaningful.